ui/static/js/fortio_chart.js

```diff
@@ -439,3 +440,26 @@ function makeMultiChart(data) {
         }
       })
 }
+
+function runTestForDuration(durationInSeconds) {
+    document.getElementById('progressBarDiv').style.height = '1em'
+    document.getElementById('progressBarDiv').style.backgroundColor = 'black'
+    var startTimeMillis = Date.now()
+    var updatePercentage = null
+
+    updatePercentage = function () {
+	var barPercentage = 0
+	if (durationInSeconds < 0) { // infinite
+	    barPercentage = (Date.now() - startTimeMillis)/20 % 100 // 2 seconds
+	} else {
+	    barPercentage = Math.min(100, (Date.now() - startTimeMillis)/(10*durationInSeconds))
+	}
+
+	document.getElementById('progressBarDiv').style.width = barPercentage + "%"
+	if (barPercentage < 100) {
+	    setTimeout(updatePercentage, 60/100)
```

Comment:
not familiar, what's the unit of 60/100 ?

Reply:
I think I meant 1000/60 :) Changed to 10 ms - still looks smooth